subworkflows/local/check_input.nf

```diff
@@ -0,0 +1,87 @@
+//
+// Check input samplesheet and get read, sample, and case channels
+//
+
+include { SAMPLESHEET_CHECK } from '../../modules/local/samplesheet_check'
+
+workflow CHECK_INPUT {
+    take:
+        ch_samplesheet // channel: [mandatory] [ path(csv) ]
+
+    main:
+        SAMPLESHEET_CHECK ( ch_samplesheet )
+            .csv
+            .splitCsv ( header:true, sep:',' )
+            .set { sheet }
+
+        case_info = sheet.first()
+                        .map { create_case_channel(it) }
+        reads     = sheet.map { row -> [[row.sample.split('_')[0]], row] }
+                        .groupTuple()
+                        .map { meta, rows ->
+                            [rows, rows.size()]
+                        }
+                        .transpose()
+                        .map { row, numLanes ->
+                            create_fastq_channel(row + [num_lanes:numLanes])
+                        }
+        samples   = sheet.map { create_samples_channel(it) }
+
+    emit:
+        case_info       // channel: [ val(case_info) ]
+        reads           // channel: [ val(meta), [ path(reads) ] ]
+        samples         // channel: [ val(sample_id), val(sex), val(phenotype), val(paternal_id), val(maternal_id), val(case_id) ]
+        versions  = SAMPLESHEET_CHECK.out.versions  // channel: [ path(versions.yml) ]
+}
+
+// Function to get list of [ meta, [ fastq_1, fastq_2 ] ]
+def create_fastq_channel(LinkedHashMap row) {
+    // create meta map
+    def meta        = [:]
+    meta.case_id    = row.case_id
+    meta.gender     = row.gender
+    meta.id         = row.sample
+    meta.maternal   = row.maternal_id
+    meta.paternal   = row.paternal_id
+    meta.phenotype  = row.phenotype
+    meta.single_end = row.single_end.toBoolean()
+    meta.num_lanes  = row.num_lanes
+    meta.read_group = "\'@RG\\tID:"+ row.fastq_1.split('/')[-1] + "\\tPL:ILLUMINA\\tSM:"+row.sample.split('_')[0]+"\'"
+
+
+    // add path(s) of the fastq file(s) to the meta map
+    def fastq_meta = []
+    if (!file(row.fastq_1).exists()) {
+        exit 1, "ERROR: Please check input samplesheet -> Read 1 FastQ file does not exist!\n${row.fastq_1}"
```

Comment:
Please use "error" instead to avoid any weir side effects

Reply:
That's odd because I had fixed it a while ago. Perhaps github did something weird. Could you check again? Because this is what I have on dev https://github.com/nf-core/raredisease/blob/1bc697a2ba2ab66163f6d7b936cb9cd4064da932/subworkflows/local/check_input.nf#L55-L61

Reply:
weird, but you removed the whole subworkflow and now and are using the check_smaplesheet file instead?